src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -225,7 +225,7 @@ internal static ClassType GetClassType(Type type)
             }
 
             // A Type is considered a value if it implements IConvertible.
-            if (typeof(IConvertible).IsAssignableFrom(type))
+            if (typeof(IConvertible).IsAssignableFrom(type) || type == typeof(DateTimeOffset))
```

Comment:
Seems odd to single out `DateTimeOffset` in this check. @steveharter, what's the appropriate place to check this?

Reply:
This is the correct place. We just check for IConvertible as a general policy for other data types. It just so happens that currently all internal data types implement this except for DataTimeOffset.

Reply:
If we are making an exception here, do we need to handle the custom type code in `DefaultConverters.cs` (i.e. here: https://github.com/dotnet/corefx/pull/36372#discussion_r269282195), or can that just be part of the regular switch statement with a new case for `DateTimeOffset`?

Reply:
It is an exception in both cases but for different reasons. Here it is an exception because DateTimeOffset doesn't implement `IConvertible`. The other location in `DefaultConverters.cs` is an exception because DateTimeOffset is not a primitive type according to `TypeCode` enum.